packages/dds/merge-tree/src/localReference.ts

```diff
@@ -366,22 +424,27 @@ export class LocalReferenceCollection {
         }
     }
 
-    public addBeforeTombstones(...refs: Iterable<LocalReference>[]) {
+    /**
+     *
+     * @internal - this method should only be called by mergeTree
+     */
+    public addBeforeTombstones(...refs: Iterable<LocalReference | ReferencePosition>[]) {
         const beforeRefs: LocalReference[] = [];
 
         for (const iterable of refs) {
             for (const lref of iterable) {
-                // eslint-disable-next-line no-bitwise
-                if (lref.refType & ReferenceType.SlideOnRemove) {
-                    beforeRefs.push(lref);
-                    lref.segment = this.segment;
-                    lref.offset = 0;
-                    if (lref.hasRangeLabels() || lref.hasTileLabels()) {
-                        this.hierRefCount++;
+                if (lref instanceof LocalReference) {
+                    if (refTypeIncludesFlag(lref, ReferenceType.SlideOnRemove)) {
```

Comment:
Is there a way to make the API more clear that SlideOnRemove is only supported on LocalReferences? Would it make sense for setting SlideOnRemove on a marker node be an error?

Reply:
i want to get there. similar to my above comment there is quite a few places where the code handles either markers or local references, but I don't think it is always necessary. these changes are incrementally cleaning that up